push.bash

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+fail() {
+  echo "::error::$1"
+  exit 1
+}
+
+# Make sure the token isn't accidentally logged
+echo "::add-mask::${BUF_TOKEN}"
+
+if [ -z "${GITHUB_SHA}" ]; then
+  fail "the commit was not provided"
+fi
+
+if [ -z "${BUF_TOKEN}" ]; then
+  fail "a buf authentication token was not provided"
+fi
+
+if [ -z "${BUF_INPUT}" ]; then
+  fail "an input was not provided"
+fi
+
+NOT_INSTALLED_MESSAGE='buf is not installed; please add the "bufbuild/buf-setup-action" step to your job found at https://github.com/bufbuild/buf-setup-action'
+
+BUF_COMMAND="$(type -P buf)" ||   fail "$NOT_INSTALLED_MESSAGE"
```

Comment:
```suggestion
BUF_COMMAND="$(type -P buf)" || fail "$NOT_INSTALLED_MESSAGE"
```

Reply:
I'm surprised to see `type` instead of `which` - any reason why you prefer this?

Reply:
`which` has inconsistent exit codes between operating systems and `type` is a builtin, so it will always behave the same in any bash implementation.

Reply:
TIL